xonsh/commands_cache.py

```diff
@@ -100,12 +101,21 @@ def all_commands(self):
         if cache_valid:
             return self._cmds_cache
         allcmds = {}
+        cnt = 0
         for path in reversed(path_immut):
             # iterate backwards so that entries at the front of PATH overwrite
             # entries at the back.
             for cmd in executables_in(path):
+                cnt += 1
                 key = cmd.upper() if ON_WINDOWS else cmd
                 allcmds[key] = (os.path.join(path, cmd), alss.get(key, None))
+
+        if cnt > builtins.__xonsh__.env.get("XONSH_WARNING_PATH_FILES", 10000):
```

Comment:
You don't need to have the `10000`, since the environment knows about its default values

Reply:
But tests fails because it uses this function without environ.

Reply:
Then you should add the environment variable to the tests :wink:

Reply:
A-ha ;)
But:
```
sphinx-build -b html -d _build/doctrees   . _build/html

Running Sphinx v3.2.1

/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/cloud_sptheme/__init__.py:79: DeprecationWarning: get_theme_dir() is deprecated, and will be removed in version 2.0

  DeprecationWarning, stacklevel=1)

Exception occurred:

  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/xonsh/commands_cache.py", line 112, in all_commands

    if cnt > builtins.__xonsh__.env.get("XONSH_WARNING_PATH_FILES"):

TypeError: '>' not supported between instances of 'int' and 'NoneType'

The full traceback has been saved in /tmp/sphinx-err-8ufyf6qk.log, if you want to report the issue to the developers.

Please also report this if it was a user error, so that a better error message can be provided next time.

A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!

Makefile:39: recipe for target 'html' failed
```
I think it's not good idea to put this env variable to the Makefile.

Reply:
Agreed, you should put them here if needed: https://github.com/xonsh/xonsh/blob/master/tests/tools.py#L109

Reply:
I've reverted all additions in tests because I think minor variable shouldn't be everywhere.

Reply:
OK, you should still remove the default here though